core/Period/Range.php

```diff
@@ -503,10 +503,17 @@ public static function getDateXPeriodsAgo($subXPeriods, $date = false, $period =
      */
     public static function getRelativeToEndDate($period, $lastN, $endDate, $site)
     {
-        $last30Relative = new Range($period, $lastN, $site->getTimezone());
-        $last30Relative->setDefaultEndDate(Date::factory($endDate));
-        $date = $last30Relative->getDateStart()->toString() . "," . $last30Relative->getDateEnd()->toString();
+        $timezone = $site->getTimezone();
+        $last30Relative = new Range($period, $lastN, $timezone);
 
+        if (strpos($endDate, '-') === false) {
```

Comment:
why do we need to apply the timezone in one case but not the other?

I would have actually expected we wouldn't need to apply a timezone here at all... especially since it doesn't seem to be used for real time / raw data. Or is it applied in case there is `today/yesterday`? Actually I think I just answered the question :) a comment be good that the timezone might be only needed for today/yesterday basically.

For some reason I would not have expected the timezone eg in `$last30Relative = new Range($period, $lastN, $timezone);` Might be a bug.

Reply:
lastN is based on today, so we'd need the timezone to figure out what 'today' is for the site?

Reply:
I see... it is used when last parameter is not set 👍